numpy/random/mtrand/mtrand.pyx

```diff
@@ -1426,15 +1435,272 @@ cdef class RandomState:
         >>> plt.show()
 
         """
+        if sizeof(long) == sizeof(npy_int64):
+            return self.random_int64(low, high, size)
+        elif sizeof(long) == sizeof(npy_int32):
+            return self.random_int32(low, high, size)
```

Comment:
I'm not sure if this will break backwards compatibility, but would logic checking the boundaries inputs be useful?  For example, they are in the range [```np.iinfo(np.int32).min, ```np.iinfo(np.int32).max```], then call ```random_int32```?  Ditto for ```random_intp```.